From 089265335dfb746a77ec15a917c12e8c77a41240 Mon Sep 17 00:00:00 2001
From: Peter Jones <pjones@redhat.com>
Date: Mon, 1 May 2017 15:34:28 -0400
Subject: [PATCH 19/22] efivar main(): explain efi_well_known_guids to the
 compiler better.

Covscan doesn't quite understand that this _is_ an array, so make it
look even more like one.

Signed-off-by: Peter Jones <pjones@redhat.com>
---
 src/efivar.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/efivar.c b/src/efivar.c
index 38d19e1..3ed9a84 100644
--- a/src/efivar.c
+++ b/src/efivar.c
@@ -461,13 +461,13 @@ int main(int argc, char *argv[])
 		case ACTION_LIST_GUIDS: {
 			efi_guid_t sentinal = {0xffffffff,0xffff,0xffff,0xffff,
 					       {0xff,0xff,0xff,0xff,0xff,0xff}};
-			extern struct guidname efi_well_known_guids;
-			extern struct guidname efi_well_known_guids_end;
+			extern struct guidname efi_well_known_guids[];
+			extern struct guidname *efi_well_known_guids_end;
 			intptr_t start = (intptr_t)&efi_well_known_guids;
 			intptr_t end = (intptr_t)&efi_well_known_guids_end;
 			unsigned int i;
 
-			struct guidname *guid = &efi_well_known_guids;
+			struct guidname *guid = &efi_well_known_guids[0];
 			for (i = 0; i < (end-start) / sizeof(*guid); i++) {
 				if (!efi_guid_cmp(&sentinal, &guid[i].guid))
 					break;
-- 
2.12.2

